### PR TITLE
{Network} `az network application-gateway ssl-policy/profile`: Migrate commands to codegen v2

### PR DESCRIPTION
**Pull request by necusjz**
**Related command**

az network application-gateway ssl-policy
az network application-gateway ssl-profile

**Description**

Total: 12 commands

**Testing Guide**

**History Notes**

[Network] BREAKING CHANGE: `az network application-gateway ssl-profile`: `verifyClientCertIssuerDn` becomes `verifyClientCertIssuerDN` in the output

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

File: src/azure-cli/azure/cli/command_modules/network/custom.py [hunk 1]
```diff
@@ -800,104 +803,96 @@ def show_ag_backend_health(cmd, resource_group_name, application_gateway_name, e
...
+class SSLProfileAdd(_SSLProfileAdd):
+    @classmethod
+    def _build_arguments_schema(cls, *args, **kwargs):
+        from azure.cli.core.aaz import AAZStrArg, AAZListArg, AAZResourceIdArg, AAZResourceIdArgFormat
+        args_schema = super()._build_arguments_schema(*args, **kwargs)
+        args_schema.client_auth_config = AAZStrArg(
+            options=["--client-auth-configuration", "--client-auth-config"],
+            help="Client authentication configuration of the application gateway resource.",
+            enum={"True": "True", "False": "False"},
         )
```
Comment (kairu-ms) [inline src/azure-cli/azure/cli/command_modules/network/custom.py:815]: Why not use AAZBoolArg?

File: src/azure-cli/azure/cli/command_modules/network/custom.py [hunk 2]
```diff
@@ -800,104 +803,96 @@ def show_ag_backend_health(cmd, resource_group_name, application_gateway_name, e
...
+        args_schema.auth_configuration._registered = False
+        args_schema.client_certificates._registered = False
+        return args_schema
 
-    return sdk_no_wait(no_wait, ncf.application_gateways.begin_create_or_update, resource_group_name,
-                       application_gateway_name, appgw)
+    def pre_operations(self):
+        args = self.ctx.args
+        if has_value(args.client_auth_config):
+            args.auth_configuration = {"verify_client_cert_issuer_dn": args.client_auth_config == "True"}
```
Comment (kairu-ms) [inline src/azure-cli/azure/cli/command_modules/network/custom.py:875]: When args.client_auth_config is null, "verify_client_cert_issuer_dn" should be cleaned.

File: src/azure-cli/azure/cli/command_modules/network/custom.py [hunk 3]
```diff
@@ -800,104 +803,96 @@ def show_ag_backend_health(cmd, resource_group_name, application_gateway_name, e
...
+    @classmethod
+    def _build_arguments_schema(cls, *args, **kwargs):
+        from azure.cli.core.aaz import AAZStrArg, AAZListArg, AAZResourceIdArg, AAZResourceIdArgFormat
+        args_schema = super()._build_arguments_schema(*args, **kwargs)
+        args_schema.client_auth_config = AAZStrArg(
+            options=["--client-auth-configuration", "--client-auth-config"],
+            help="Client authentication configuration of the application gateway resource.",
+            enum={"True": "True", "False": "False"},
+            nullable=True,
+        )
```
Comment (kairu-ms) [inline src/azure-cli/azure/cli/command_modules/network/custom.py:855]: Recommend using AAZBoolArg

File: src/azure-cli/azure/cli/command_modules/network/tests/latest/test_network_commands.py [hunk 4]
```diff
@@ -1037,10 +1037,10 @@ def test_network_app_gateway_with_ssl_profile(self, resource_group):
 
         self.cmd('network application-gateway ssl-profile update -g {rg} --gateway-name {gw} --name {name1} '
                  '--client-auth-configuration False',
-                 checks=[self.check('sslProfiles[1].clientAuthConfiguration.verifyClientCertIssuerDn', False)])
+                 checks=[self.check('sslProfiles[1].clientAuthConfiguration.verifyClientCertIssuerDN', False)])
```
Comment (kairu-ms) [inline src/azure-cli/azure/cli/command_modules/network/tests/latest/test_network_commands.py:1040]: It's a breaking change in output. Please add this in the release notes.

**Comment by yonzhan** [comment]
Network

**Review by kairu-ms — APPROVED** [review]

**Pull request merged**